### PR TITLE
Feat/2/Statement Managment

### PR DESCRIPTION
**Pull request by NataliaHerus**
Tech-dept solution of implementing statement managment

### CODE REVIEW

File: MilitaryManager/client-app/src/app/shared/state.service.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  protected setState(newState: Partial<T>) {
+    this.state$.next({
+      ...this.state,
+      ...newState,
+    });
+  }
+
+}
```
Comment (vladtymo) [inline MilitaryManager/client-app/src/app/shared/state.service.ts]: Add the line break;

**Review by vladtymo — APPROVED** [review]

File: MilitaryManager/client-app/src/app/shared/services/attachments-state.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const initialState: AttachmentState = {
+  attachments: []
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AttachmentStateService extends StateService<AttachmentState>{
+
+  units$: Observable<AttachmentModel[]>= this.select(state => state.attachments).pipe(
```
Comment (Zakerman110) [inline MilitaryManager/client-app/src/app/shared/services/attachments-state.service.ts]: Rename `units$` to `attachments$`

File: MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts [hunk 3]
```diff
@@ -4,12 +4,23 @@ import { HttpService } from "../core/http.service";
...
+  
+  http: HttpService;
   constructor(
     httpService: HttpService,
     configService: ClientConfigurationService) {
     super(httpService, 'weatherForecast', configService, UnitModel, ServiceType.units);
+    this.http = httpService;
   }
+
+  getAll(): Observable<UnitModel[]> {
```
Comment (Zakerman110) [inline MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts]: No need to create `getAll()` method, it is already implemented in BaseService. As a result not need for 14th and 19th lines

File: MilitaryManager/client-app/src/app/shared/services/unit-state.service.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  units$: Observable<UnitModel[]>= this.select(state => state.units).pipe(
+    shareReplay({refCount: true, bufferSize: 1}))
+
+  constructor(private apiService: UnitsService) {
+    super(initialState);
+  }
+  
+  load() {
+    this.apiService.getAll().subscribe((units) => this.setState({ units}));
```
Comment (Zakerman110) [inline MilitaryManager/client-app/src/app/shared/services/unit-state.service.ts]: You can use already implemented method `this.apiService.collection.getAll().subscribe((units) => this.setState({ units}));`

**Review by Zakerman110 — CHANGES_REQUESTED** [review]

**Review by Zakerman110 — APPROVED** [review]

File: MilitaryManager.Attachments.API/Properties/launchSettings.json [hunk 5]
```diff
@@ -21,16 +21,17 @@
...
       "commandName": "Docker",
       "launchBrowser": true,
       "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}/weatherforecast",
-      "environmentVariables": {}
+      "publishAllPorts": true,
+      "useSSL": true
     }
   }
-}
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.Attachments.API/Properties/launchSettings.json]: End of file

File: MilitaryManager.IdentityServer/Controllers/AccountController.cs [hunk 6]
```diff
@@ -122,19 +122,24 @@ public async Task<IActionResult> Register(RegisterViewModel model, string return
 				{
 					UserName = model.Email,
 					Email = model.Email,
+					//Roles =
+					//{
+					//	new IdentityUserRole<string>
+					//	{
+					//		RoleId = (await _roleManager.FindByNameAsync(Roles.User.ToString())).Id
+					//	}
+					//}
```
Comment (YevheniiVovk) [inline MilitaryManager.IdentityServer/Controllers/AccountController.cs]: Why it commented ?

File: MilitaryManager.IdentityServer/IdentityConfig.cs [hunk 7]
```diff
@@ -16,7 +16,7 @@ public static IEnumerable<IdentityResource> GetIdentityResources()
 				new IdentityResource("role", new []{ JwtClaimTypes.Role })
 			};
 		}
-
+
```
Comment (YevheniiVovk) [inline MilitaryManager.IdentityServer/IdentityConfig.cs]: Empty line with spaces

File: MilitaryManager.IdentityServer/Program.cs [hunk 8]
```diff
@@ -13,39 +8,6 @@ public class Program
...
-                    };
-
-                    var result = userManager.CreateAsync(user, PASSWORD).Result;
-                    if (result.Succeeded)
-                    {
-                        userManager.AddToRoleAsync(user, "Admin");
-                    }
-                }
-            }
-
```
Comment (YevheniiVovk) [inline MilitaryManager.IdentityServer/Program.cs]: Why did you remove this and actually how it connect with your task

File: MilitaryManager.Attachments.API/Properties/launchSettings.json (hunk 5 above)
Comment (YevheniiVovk) [inline MilitaryManager.Attachments.API/Properties/launchSettings.json]: Empty line

File: MilitaryManager.IdentityServer/Properties/launchSettings.json [hunk 9]
```diff
@@ -18,16 +18,17 @@
...
       "commandName": "Docker",
       "launchBrowser": true,
       "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}",
-      "environmentVariables": {}
+      "publishAllPorts": true,
+      "useSSL": true
     }
   }
-}
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.IdentityServer/Properties/launchSettings.json]: Empty line

File: MilitaryManager.Units.API/Properties/launchSettings.json [hunk 10]
```diff
@@ -21,16 +21,17 @@
...
       "commandName": "Docker",
       "launchBrowser": true,
       "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}/weatherforecast",
-      "environmentVariables": {}
+      "publishAllPorts": true,
+      "useSSL": true
     }
   }
-}
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.Units.API/Properties/launchSettings.json]: Empty line

File: MilitaryManager/Properties/launchSettings.json [hunk 11]
```diff
@@ -18,16 +18,17 @@
...
       "commandName": "Docker",
       "launchBrowser": true,
       "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}",
-      "environmentVariables": {}
+      "publishAllPorts": true,
+      "useSSL": true
     }
   }
-}
+}
```
Comment (YevheniiVovk) [inline MilitaryManager/Properties/launchSettings.json]: Empty line

File: docker-compose.override.yml [hunk 12]
```diff
@@ -44,4 +44,4 @@ services:
 
     volumes:
       - ${APPDATA}/Microsoft/UserSecrets:/root/.microsoft/usersecrets:ro
-      - ${APPDATA}/ASP.NET/Https:/root/.aspnet/https:ro
+      - ${APPDATA}/ASP.NET/Https:/root/.aspnet/https:ro
```
Comment (YevheniiVovk) [inline docker-compose.override.yml]: Empty line

**Pull request merged**